src/analysis/utility/analytics.py

```diff
@@ -0,0 +1,213 @@
+import pymannkendall as mk
+import networkx as nx
+import json
+import numpy as np
+from sklearn.linear_model import LinearRegression
+import seaborn as sns
+import matplotlib.pyplot as plt
+from analysis.utility.charting import *
+import statsmodels.api as sm
+from statsmodels.stats.diagnostic import linear_rainbow
+import ruptures as rpt
+
+
+
+def checkIfTrendIsRising(df, metric):
+    # A little bit of regression and then check slope
+    # We may have to change boolean to a value
+    
+    #Run Mann-Kendall Test. It is a non parametric test
+    
+    trend = False
+    # Perform the Mann-Kendall test
+    result = mk.original_test(df[metric])
+
+    #print(f"Mann-Kendall Test Statistic: {result.z}")
+    #print(f"P-value: {result.p}")
+    if result.p < 0.05 and result.trend == 'increasing':
+        trend = True
+        print("There is a statistically significant rising trend in: ",metric)
+    else:
+        print("No significant trend detected in: ",metric)
+    
+    return trend
+
+#0.5 sec threshold default
+def pctTimeValueIsHigh(df, metric,thresholdValue = 0.5):
+    # How many times have this exceeded over a given value
+    # How many observations in total
+    # Therefore conclude ....
+    
+    countThresholhExceed = (df[metric] > thresholdValue).sum() 
+    totalCount = len(df[metric])
+    #print(thresholdValue)
```

Comment:
We can probably just have debug logs for this and other print statements, and only print out results?